core/src/main/scala/org/http4s/blaze/pipeline/stages/SSLStage.scala

```diff
@@ -182,15 +186,15 @@ final class SSLStage(engine: SSLEngine, maxWrite: Int = 1024 * 1024)
           o.flip()
 
           if (r.bytesProduced < 1)
-            logger.warn(s"SSL Handshake WRAP produced 0 bytes.\n$r")
+            handshakeFailure(new SSLException(s"SSL Handshake WRAP produced 0 bytes: $r"))
```

Comment:
This is a loop we're getting into in http4s/http4s#1492. Let's see what happens if we fail it aggressively.